libraries/lib-stretching-sequence/SilenceSegment.cpp

```diff
@@ -0,0 +1,37 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  SilenceSegment.cpp
+
+  Matthieu Hodgkinson
+
+**********************************************************************/
+#include "SilenceSegment.h"
+
+#include <algorithm>
+
+SilenceSegment::SilenceSegment(size_t numChannels, sampleCount numSamples)
+    : mNumChannels(numChannels)
+    , mNumRemainingSamples(numSamples)
+{
+}
+
+size_t SilenceSegment::GetFloats(float* const* buffers, size_t numSamples)
+{
+   const size_t numSamplesToProduce =
+      std::min<long long>(mNumRemainingSamples.as_long_long(), numSamples);
+   for (auto i = 0u; i < mNumChannels; ++i)
+   {
+      auto buffer = buffers[i];
+      std::fill(buffer, buffer + numSamplesToProduce, 0.f);
+   }
+   mNumRemainingSamples -= numSamplesToProduce;
+   return numSamplesToProduce;
```

Comment:
Only now do I understand the meaning of the return value of `GetFloats`.  But AudioSegment.h, lacking comments, didn't help me.

Please write some informal description of the contract in that header file.

If it were me, I would even add a pure virtual `size_t GetWidth() const` just so that I could use it to state a precondition on the size of `buffers` and call it just in not-NDEBUG builds to assert the precondition.